### PR TITLE
Use Redux in React components

### PR DESCRIPTION
**Pull request by SelmaNdapanda**
## In this pull request, I implemented the following:

- I made sure that I kept the state immutable in reducers!
- I changed the default state in my books reducer from an empty array to an array with a few books. 
- I imported Redux Provider and my store into the main component of your React app.
- I connected my app to the Redux store.
- I made sure that the list of books displays books from the Redux store.
- I added functionality for the "Add new" button click in my React component responsible for adding new books.
- I added functionality for the "Remove" button click in my React component responsible for removing books.
- I added functionality for the "Check status" button click in my React component responsible for categories.

### CODE REVIEW

File: src/components/Bookitem.js [hunk 1]
```diff
@@ -2,14 +2,14 @@ import React from 'react';
...
-  const { title, author, deleteBook } = props;
+  const { title, author, remove } = props;
 
   return (
     <div className="container">
       <div className="bookCard">
         <p className="title">{title}</p>
         <p className="author">{author}</p>
-        <button type="button" className="Remove" onClick={deleteBook}>Remove</button>
+        <button type="button" className="Remove" onClick={remove}>Remove</button>
```
Comment (prolajumokeoni) [inline src/components/Bookitem.js:12]: - Some of the[ requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components.md#project-requirements) below are not yet implemented in this component ⬇️ 

```
In your React component responsible for removing books, add a functionality for "Remove" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).
```

- [x]  Kindly make sure that you import the action creator in your redux  and use it in creating the function for removing the book 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @SelmaNdapanda, 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used 
✔️  Good PR title and Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Bookitem.js (hunk 1 above)
Comment (SelmaNdapanda) [reply src/components/Bookitem.js:12]: Hi @prolajumokeoni , Thank you for the review. But I did that already in the Book.js file

**Review by ShadyShawkat — APPROVED** [review]
## Hi @SelmaNdapanda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**